tests/devices/test_default_qutrit.py

```diff
@@ -260,6 +260,292 @@ def test_apply_rotations_one_wire(self, qutrit_device_1_wire, tol):
     # TODO: Add tests for state preperation ops after they're implemented
 
 
+class TestExpval:
+    """Tests that expectation values are properly calculated or that the proper errors are raised."""
+
+    # TODO: Add test for expval of non-parametrized observables
+
+    @pytest.mark.parametrize(
+        "observable,state,expected_output,mat",
+        [
+            (qml.THermitian, [1, 0, 0], 1, [[1, 1j, 0], [-1j, 1, 0], [0, 0, 1]]),
+            (qml.THermitian, [0, 1, 0], -1, [[1, 0, 0], [0, -1, 0], [0, 0, 0]]),
+            (
+                qml.THermitian,
+                [1 / math.sqrt(3), -1 / math.sqrt(3), 1j / math.sqrt(3)],
+                0,
+                [[0, -1j, 0], [1j, 0, 0], [0, 0, 0]],
+            ),
+        ],
+    )
+    def test_expval_single_wire_with_parameters(
+        self, qutrit_device_1_wire, tol, observable, state, expected_output, mat
+    ):
+        """Tests that expectation values are properly calculated for single-wire observables with parameters."""
+
+        obs = observable(np.array(mat), wires=[0])
+
+        qutrit_device_1_wire.reset()
+        qutrit_device_1_wire._state = np.array(state).reshape([3])
+        qutrit_device_1_wire.apply([], obs.diagonalizing_gates())
+        res = qutrit_device_1_wire.expval(obs)
+
+        assert np.isclose(res, expected_output, atol=tol, rtol=0)
+
+    A = np.array([[1, 0, 0], [0, -1, 0], [0, 0, 2]])
+    B = np.array([[4, 0, 0], [0, -2, 0], [0, 0, 1]])
+    obs_1 = np.kron(A, B)
+
+    C = np.array([[0, -1j, 0], [1j, 0, 0], [0, 0, 0]])
+    D = np.array([[1, 2, 3], [2, 1, 3], [3, 3, 2]])
+    obs_2 = np.kron(C, D)
+
+    @pytest.mark.parametrize(
+        "observable,state,expected_output,mat",
+        [
+            (
+                qml.THermitian,
+                [1 / math.sqrt(3), 0, 1 / math.sqrt(3), 1 / math.sqrt(3), 0, 0, 0, 0, 0],
+                1 / 3,
+                obs_1,
+            ),
+            (
+                qml.THermitian,
+                [0, 0, 0, 0, 0, 0, 0, 0, 1],
+                2,
+                obs_1,
+            ),
+            (
+                qml.THermitian,
+                [0.5, 0, 0, 0.5, 0, 0, 0, 0, 1 / math.sqrt(2)],
+                1,
+                obs_1,
+            ),
+            (
+                qml.THermitian,
+                [
+                    3.73671170e-01 - 0.00000000e00j,
```

Comment:
Is this one randomly-generated?

Reply:
I don't recall how I created this state 😅